adoc/extensions/sycl_khr_free_function_commands.adoc

```diff
@@ -0,0 +1,928 @@
+[code]#DestMode# must be [code]#access_mode::write# or
+[code]#access_mode::read_write#.
+
+_Preconditions_: [code]#src# must be a host pointer, pointing to an allocation
+of at least as many bytes as the range represented by [code]#dest#.
+
+_Effects (3-4)_: Equivalent to [code]#h.copy(src, dest)#.
+
+_Effects (5-6)_: Equivalent to [code]#q.submit([&](handler& h) { copy(h, src,
+dest) })#
+
+'''
+
+.[apititle]#copy# (accessors, device to host)
+[source,role=synopsis,id=api:copy-accessor-d2h]
+----
+namespace sycl::khr {
+
+template <typename SrcT, int SrcDims, access_mode SrcMode, typename DestT>
+void copy(sycl::handler& h,
+          sycl::accessor<SrcT, SrcDims, SrcMode, target::device> src,
+          DestT* dest);                 (7)
+
+template <typename SrcT, int SrcDims, access_mode SrcMode, typename DestT>
+void copy(sycl::handler& h,
+          sycl::accessor<SrcT, SrcDims, SrcMode, target::device> src,
+          std::shared_ptr<DestT> dest); (8)
+
+template <typename SrcT, int SrcDims, access_mode SrcMode, typename DestT>
+void copy(sycl::queue q,
+          sycl::accessor<SrcT, SrcDims, SrcMode, target::device> src,
+          DestT* dest);                 (9)
+
+template <typename SrcT, int SrcDims, access_mode SrcMode, typename DestT>
+void copy(sycl::queue q,
+          sycl::accessor<SrcT, SrcDims, SrcMode, target::device> src,
+          std::shared_ptr<DestT> dest); (10)
+
+}
+----
+
+Copies from device to host.
+
+_Constraints_: [code]#SrcT# and [code]#DestT# must be <<device-copyable>>.
+[code]#DestMode# must be [code]#access_mode::read# or
+[code]#access_mode::read_write#.
+
+_Preconditions_: [code]#dest# must be a host pointer, pointing to an allocation
+of at least as many bytes as the range represented by [code]#src#.
+
+_Effects (7-8)_: Equivalent to [code]#h.copy(src, dest)#.
+
+_Effects (9-10)_: Equivalent to [code]#q.submit([&](handler& h) { copy(h, src,
+dest); })#.
+
+'''
+
+.[apititle]#copy# (accessors, device to device)
+[source,role=synopsis,id=api:copy-accessor-d2d]
+----
+namespace sycl::khr {
+
+template <typename SrcT, int SrcDims, access_mode SrcMode,
+          typename DestT, int DestDims, access_mode DestMode>
+void copy(sycl::queue q,
+          sycl::accessor<SrcT, SrcDims, SrcMode, target::device> src,
+          sycl::accessor<DestT, DestDims, DestMode, target::device> dest); (11)
+
+template <typename SrcT, int SrcDims, access_mode SrcMode,
+          typename DestT, int DestDims, access_mode DestMode>
+void copy(sycl::queue q,
+          sycl::accessor<SrcT, SrcDims, SrcMode, target::device> src,
+          sycl::accessor<DestT, DestDims, DestMode, target::device> dest); (12)
+
+}
+----
+
+Copies between two device accessors.
+
+_Constraints_: [code]#SrcT# and [code]#DestT# must be <<device-copyable>>.
+[code]#SrcMode# must be [code]#access_mode::read# or
+[code]#access_mode::read_write#.
+[code]#DestMode# must be [code]#access_mode::write# or
+[code]#access_mode::read_write#.
```

Comment:
ISO C++ is not consistent about this, but they seem to use a bulleted list when there are many constraints.  For example, see [here](https://eel.is/c++draft/containers#flat.map.modifiers-19).  Again, I've been adopting this style in #670 and the related "vec" PRs.  I'm using a bulleted list and the word "and" like:

> _Constraints:_
> * `SrcT` and `DestT` are device-copyable,
> * `SrcMode` is `access_mode::read` or `access_mode::read_write`, and
> * `DestMode` is `access_mode::write` or `access_mode::read_write`.

Reply:
Ooh.  I like this much more; adopted in 47c08f8623d9be26f7116fd9da71fa42ba71aa02.